llvm/utils/TableGen/SubtargetEmitter.cpp

```diff
@@ -260,7 +260,7 @@ unsigned SubtargetEmitter::FeatureKeyValues(
   llvm::sort(FeatureList, LessRecordFieldName());
 
   // Check that there are no duplicate keys
-  std::set<StringRef> UniqueKeys;
+  llvm::StringSet<> UniqueKeys;
```

Comment:
What's the advantage of this?

Reply:
My understanding is that unordered containers are generally more efficient, and in this case we do not iterate over the set or map, so ordering is not required. That's the motivation. I tried unorderd_set<StringRef> but that did not work, so switched to StringSet<>.

Reply:
This should use `DenseSet<StringRef>`, not `StringSet`. Only use StringSet if the set must own the strings.

Reply:
Thanks. It wasn't immediately clear, but I now see that https://llvm.org/docs/ProgrammersManual.html#llvm-adt-stringmap-h mentions that: "The string data (key) and the element object (value) are stored in the same allocation". I will change it to DenseSet, Also, I am adding an addition enhancement to track the previous definition so its easy to see where the 2 duplicate definitions are.